package/kedro_viz/integrations/kedro/sqlite_store.py

```diff
@@ -55,7 +77,102 @@ def to_json(self) -> str:
         return json.dumps(session_dict)
 
     def save(self):
-        """Save the session store info on db ."""
-
+        """Save the session store info on db and uploads it
+        to the cloud if a remote cloud path is provided ."""
         with self._db_session_class.begin() as session:
-            session.add(RunModel(id=self._session_id, blob=self.to_json()))
+            session.add(RunModel(id=self._session_id, blob=self._to_json()))
+        if self.remote_location:
+            self._upload()
+
+    def _upload(self):
+        """Uploads the session store database file to the specified
+        remote path on the cloud storage."""
+        db_name = _get_dbname()
+        logger.debug(f"Uploading local session store to remote with name {db_name}...")
+        try:
+            self._remote_fs.put(self.location, f"{self.remote_location}/{db_name}")
+        except Exception as exc:
+            logger.exception(exc)
+
+    def _download(self):
+        """Downloads all the session store database files
+        from the specified remote path on the cloud storage
+        to your local project.
+        """
+        try:
+            # In theory we should be able to do this as a single operation:
+            # self._remote_fs.get(f"{self.remote_location}/*.db", str(Path(self.location).parent))
+            # but this does not seem to work correctly - maybe a bug in fsspec. So instead
+            # we do it in two steps. Also need to add os.sep so it works with older s3fs version.
+            # This is a known bug in s3fs
+            remote_dbs = self._remote_fs.glob(f"{self.remote_location}/*.db")
+            logger.debug(
+                f"Downloading {len(remote_dbs)} remote session stores to local..."
+            )
+            for remote_db in remote_dbs:
+                self._remote_fs.get(remote_db, str(Path(self.location).parent) + os.sep)
+        except Exception as exc:
+            logger.exception(exc)
```

Comment:
Similar comment as for the upload method: I think it would be useful to mention that this exception happens in `download` so it's easier to trace.